packages/metro-config/index.js

```diff
@@ -17,132 +17,24 @@
-}
-
-function build(projectRoot, moreExtraNodeModules = {}) {
-  const linkedDependencies = getLinkedDependencies(projectRoot);
-  console.log(`Linked dependencies: ${linkedDependencies.map(({ name }) => name).join(", ")}`);
-
-  const watchFolders = linkedDependencies.map((pkg) => pkg.path);
-
-  const blockedPaths = [];
-  for (const pkg of linkedDependencies) {
-    // Block the links
-    blockedPaths.push(path.join(projectRoot, "node_modules", pkg.name));
-    // Block any peer dependencies of the links
-    blockedPaths.push(...pkg.peerDependencies.map((peer) => path.join(pkg.path, "node_modules", peer)));
-  }
-  // Turn paths into regular expressions
-  const blockList = blockedPaths.map((pkgPath) => new RegExp(`^${escape(pkgPath)}\\/.*$`));
-
-  const extraNodeModulesBase = {
-    ...moreExtraNodeModules,
-  };
-  const allPeerDependencies = linkedDependencies.flatMap((pkg) => pkg.peerDependencies);
-  const unresolvedPeerDependencies = new Set(allPeerDependencies.filter((name) => !(name in linkedDependencies)));
-  // Locate the package providing every unresolved peer dependency
-  for (const peer of unresolvedPeerDependencies) {
-    for (const pkg of linkedDependencies) {
-      const potentialPath = path.join(pkg.path, "node_modules", peer);
-      if (fs.existsSync(potentialPath)) {
-        extraNodeModulesBase[peer] = potentialPath;
-        break;
-      }
-    }
-  }
-  // Create a proxy around the resolved peer dependencies
-  // This will resolve all modules exposed by the react-native app and fall back to resolved peer dependencies
-  const extraNodeModules = new Proxy(extraNodeModulesBase, {
-    get(target, name) {
-      const candidatePaths = [];
-      const potentialPath = path.join(projectRoot, "node_modules", name);
-      if (fs.existsSync(potentialPath)) {
-        candidatePaths.push(potentialPath);
-      }
-      if (name in target) {
-        candidatePaths.push(target[name]);
-      }
-      // Try resolving this from the linked dependencies
-      const linkedDependencyDependencies = linkedDependencies
-        // Skipping packages that are devDependencies, since they shouldn't be relied on
-        .filter((pkg) => !pkg.devDependencies.includes(name))
-        .map((pkg) => path.join(pkg.path, "node_modules", name))
-        .filter(fs.existsSync);
-      candidatePaths.push(...linkedDependencyDependencies);
-
-      if (candidatePaths.length === 1) {
-        return candidatePaths[0];
-      } else if (candidatePaths.length > 1) {
-        if (DEBUG_METRO_CONFIG) {
-          console.warn(`Found multiple '${name}' packages via linked dependencies:`);
-          for (const i in candidatePaths) {
-            console.warn(`- ${candidatePaths[i]}`, i === 0 ? " (returning this)" : "");
-          }
-        }
-        return candidatePaths[0];
-      } else {
-        console.error("Failed to resolve", name);
-        return undefined;
-      }
+function build({ projectRoot, watchFolders }) {
```

Comment:
Did `workspaces` fix this complexity?  Or is this just a new discovery for making the symbolic links problem go away in a simpler way?

Reply:
It was possible to really simplify this, which is great! 😄 👍

Reply:
That IS great! 👍🏼